main/Main.hs

```diff
@@ -358,12 +358,12 @@ reReadConfig startingUp pool gucConfigEnabled env path refConf dbUriFile secretF
     Left err   ->
       if startingUp
         then panic err -- die on invalid config if the program is starting up
-        else hPutStrLn stderr $ "Failed config load. " <> err
+        else hPutStrLn stderr $ "Failed loading in-database config. " <> err
```

Comment:
The term in-database config is much clearer.
Perhaps the `db-load-guc-config` option should be changed to `config-in-db`?

Reply:
Hm, I see your point about `db-load-guc-config` - it's clumsy. I'd like to keep the `db-` prefix, though.

What about `db-has-config = true|false` ? Or just `db-config`.

Reply:
The `has` looks a bit weird inside a config option. I'll go with `db-config`, it's already much better than `db-load-guc-config`.

Reply:
@wolfgangwalther What do you think of having `db-config` to be `false` by default

The main argument for it being `true` was to have the least amount of config options: https://github.com/PostgREST/postgrest/pull/1729#issuecomment-759564831. But that isn't possible atm, because `db-anon-role` and `db-schemas` are required.

However I'm thinking that only defining `db-uri` could be done and still maintain the required fields restriction. This by having an `DbUriAppConfig` that is used previous to `AppConfig`. With that even non-reloadable config options(like `db-pool`) could be set with the `db-config`. 

But anyway the above is still not done, so `db-config` is not an advantage now regarding few config options.

It's just that I feel `db-config=false` is right because most users will not use it(or won't be able, as mentioned on https://github.com/PostgREST/postgrest/pull/1729#issuecomment-758751747). It's a special use case. Having it by default gives potential for errors(because of the query done, twice) on all instances.

WDYT?

Reply:
> db-anon-role and db-schemas are required.
> This by having an DbUriAppConfig that is used previous to AppConfig

Actually, a better solution for this would be to:

- make `db-anon-role` optional. This has some value as we discussed before(some users want only JWT access).
- make `db-schemas` default to `public`. This seems like a nice default that will always work.

Reply:
> > db-anon-role and db-schemas are required.
> > This by having an DbUriAppConfig that is used previous to AppConfig
> 
> Actually, a better solution for this would be to:
> 
>     * make `db-anon-role` optional. #1689
> 
>     * make `db-schemas` default to `public`. This seems like a nice default that will always work. We can put a message on stdout to make it clear when the default is used.

Much, much better, yes!

Reply:
> It's just that I feel `db-config=false` is right because most users will not use it(or won't be able, as mentioned on [#1729 (comment)](https://github.com/PostgREST/postgrest/pull/1729#issuecomment-758751747)). It's a special use case. Having it by default gives potential for errors(because of the query done, twice) on all instances.
> 
> WDYT?

I think the risk involved is very small. There is no overhead either - even though it's a separate statement, it's basically just part of reloading the schema cache. And in the big picture, loading config options from the database is just a tiny piece of reading the whole schema.

tbh: I actually wonder whether we really need the config option at all. It could just be enabled all the time. It's probably most useful for writing tests - but that's about it.

Reply:
> And in the big picture, loading config options from the database is just a tiny piece of reading the whole schema.

Ah, you're totally right here! I remember trying to do this at first, but I had some issues with the code.

I think it could be done now. We'd have to enable reloading only a part of the schema cache(the guc settings) for SIGUSR2/NOTIFY. Also make the `--dump-config` wait for the `connectionWoker` to finish(maybe an MVar) and only print the config contents to stdout(no `Attempt to connect to db..` or tests fail).

> tbh: I actually wonder whether we really need the config option at all. It could just be enabled all the time. 

I think the same way now. If the guc settings are part of the schema cache, we can remove the config option.